x-pack/plugins/security_solution/server/lib/detection_engine/rule_monitoring/logic/rule_execution_log/client_for_executors/client.ts

```diff
@@ -0,0 +1,253 @@
+import {
+  LogLevel,
+  logLevelFromExecutionStatus,
+  LogLevelSetting,
+  logLevelToNumber,
+  ruleExecutionStatusToNumber,
+} from '../../../../../../../common/detection_engine/rule_monitoring';
+
+import { assertUnreachable } from '../../../../../../../common/utility_types';
+import { withSecuritySpan } from '../../../../../../utils/with_security_span';
+import { truncateValue } from '../utils/normalization';
+import type { ExtMeta } from '../utils/console_logging';
+import { getCorrelationIds } from './correlation_ids';
+
+import type { IEventLogWriter } from '../event_log/event_log_writer';
+import type { IRuleExecutionSavedObjectsClient } from '../execution_saved_object/saved_objects_client';
+import type {
+  IRuleExecutionLogForExecutors,
+  RuleExecutionContext,
+  StatusChangeArgs,
+} from './client_interface';
+
+export const createClientForExecutors = (
+  settings: RuleExecutionSettings,
+  soClient: IRuleExecutionSavedObjectsClient,
+  eventLog: IEventLogWriter,
+  logger: Logger,
+  context: RuleExecutionContext
+): IRuleExecutionLogForExecutors => {
+  const baseCorrelationIds = getCorrelationIds(context);
+  const baseLogSuffix = baseCorrelationIds.getLogSuffix();
+  const baseLogMeta = baseCorrelationIds.getLogMeta();
+
+  const { executionId, ruleId, ruleUuid, ruleName, ruleType, spaceId } = context;
+
+  const client: IRuleExecutionLogForExecutors = {
+    get context() {
+      return context;
+    },
+
+    trace(...messages: string[]): void {
+      const message = messages.join(' ');
+      writeMessageToConsole(message, LogLevel.trace, baseLogMeta);
+      writeMessageToEventLog(message, LogLevel.trace);
+    },
+
+    debug(...messages: string[]): void {
+      const message = messages.join(' ');
+      writeMessageToConsole(message, LogLevel.debug, baseLogMeta);
+      writeMessageToEventLog(message, LogLevel.debug);
+    },
+
+    info(...messages: string[]): void {
+      const message = messages.join(' ');
+      writeMessageToConsole(message, LogLevel.info, baseLogMeta);
+      writeMessageToEventLog(message, LogLevel.info);
+    },
+
+    warn(...messages: string[]): void {
+      const message = messages.join(' ');
+      writeMessageToConsole(message, LogLevel.warn, baseLogMeta);
+      writeMessageToEventLog(message, LogLevel.warn);
+    },
+
+    error(...messages: string[]): void {
+      const message = messages.join(' ');
+      writeMessageToConsole(message, LogLevel.error, baseLogMeta);
+      writeMessageToEventLog(message, LogLevel.error);
+    },
+
+    async logStatusChange(args: StatusChangeArgs): Promise<void> {
+      await withSecuritySpan('IRuleExecutionLogForExecutors.logStatusChange', async () => {
+        const correlationIds = baseCorrelationIds.withStatus(args.newStatus);
+        const logMeta = correlationIds.getLogMeta();
+
+        try {
+          writeStatusChangeMessageToConsole(args, logMeta);
+          writeStatusChangeToConsole(args, logMeta);
```

Comment:
What's the difference between `writeStatusChangeMessageToConsole` and `writeStatusChangeToConsole`? It's not very clear from the function names.

Reply:
Discussed with @xcrzx. Decided to combine these two functions into one.